rules/integrations/aws/persistence_lambda_backdoor_invoke_function_for_any_principal.toml

```diff
@@ -74,14 +103,15 @@ tags = [
     "Resources: Investigation Guide",
 ]
 timestamp_override = "event.ingested"
-type = "query"
+type = "eql"
 
 query = '''
-event.dataset: aws.cloudtrail
-    and event.provider: lambda.amazonaws.com
-    and event.outcome: success
-    and event.action: AddPermission*
-    and aws.cloudtrail.request_parameters: (*lambda\:InvokeFunction* and *principal=\**)
+info where event.dataset == "aws.cloudtrail" 
+    and event.provider == "lambda.amazonaws.com" 
+    and event.outcome == "success" 
+    and event.action : "AddPermission*" 
+    and stringContains(aws.cloudtrail.request_parameters, "lambda:InvokeFunction") 
+    and stringContains(aws.cloudtrail.request_parameters, "principal=\\*")
```

Comment:
was this tested in match on `princioal=*` ?

Reply:
Yes I tested to capture that, I have the screenshot of the working query in the issue description. If you run it in our test stack it'll capture that event from yesterday.